src/renderer/components/+extensions/extension-install.store.ts

```diff
@@ -1,13 +1,187 @@
-import { observable } from "mobx";
-import { autobind, Singleton } from "../../utils";
+import { action, computed, observable } from "mobx";
+import logger from "../../../main/logger";
+import { disposer, Disposer } from "../../utils";
+import * as uuid from "uuid";
 
-interface ExtensionState {
-  displayName: string;
-  // Possible states the extension can be
-  state: "installing" | "uninstalling";
+export enum ExtensionInstallationState {
+  INSTALLING = "installing",
+  UNINSTALLING = "uninstalling",
+  IDLE = "IDLE",
 }
 
-@autobind()
-export class ExtensionStateStore extends Singleton {
-  extensionState = observable.map<string, ExtensionState>();
+const Prefix = "[ExtensionInstallationStore]";
+const installingExtensions = observable.set<string>();
+const uninstallingExtensions = observable.set<string>();
+const preInstallIds = observable.set<string>();
+
+export class ExtensionInstallationStateStore {
+  @action static reset() {
```

Comment:
I see only `reset` being used .. are other methods going to be used in some other PR?

Reply:
Do you check `src/renderer/components/+extensions/extensions.tsx`? It is by default not shown because it is considered "large"

Reply:
Missed that, now it makes more sense 🙂